test/stdexec/concepts/test_concepts_sender.cpp

```diff
@@ -144,6 +144,18 @@ TEST_CASE("check completion signatures for sender that also supports error codes
   check_sends_stopped<false>(ec_sender{});
   REQUIRE(ex::sender_of<ec_sender, ex::set_value_t()>);
 }
+struct my_r5_sender0 {
+  using completion_signatures =
+    ex::completion_signatures<             //
+      ex::set_value_t(),                   //
+      ex::set_error_t(std::exception_ptr), //
+      ex::set_stopped_t()>;
+
+  friend oper tag_invoke(ex::connect_t, my_r5_sender0, empty_recv::recv0&& r) { return {}; }
+};
+TEST_CASE("r5 sender emits deprecated diagnostics", "[concepts][sender]") {
+  ex::get_env(my_r5_sender0{});
+}
```

Comment:
`my_r5_sender0` isn't an R5 sender because it doesn't support `get_completion_signatures(sndr, no_env{})`.

Reply:
Hm, why not? It satisfies the `sender` concept (which as of my branch is only changed from R5 by the additional requirement of `get_env(sndr)` being valid). The `completion_signatures` member type should handle any env, including `no_env.`

Reply:
Yeah, don't listen to my. My brain is addled.